crates/router/src/core/payments/operations/payment_create.rs

```diff
@@ -1023,6 +1027,43 @@ impl PaymentCreate {
             .change_context(errors::ApiErrorResponse::InternalServerError)?
             .map(Secret::new);
 
+        // Derivation of directly supplied Billing Address data in our Payment Create Request
+        let mut raw_billing_address_details = AddressDetails::default();
+        let details_present = request.billing.clone().map(|billing_details| {
+            billing_details.address.clone().map(|address| {
+                raw_billing_address_details.set_city(address.city);
+                raw_billing_address_details.set_country(address.country);
+                raw_billing_address_details.set_line1(address.line1);
+                raw_billing_address_details.set_line2(address.line2);
+                raw_billing_address_details.set_line3(address.line3);
+                raw_billing_address_details.set_zip(address.zip);
+                raw_billing_address_details.set_state(address.state);
+                raw_billing_address_details.set_first_name(address.first_name);
+                raw_billing_address_details.set_last_name(address.last_name);
+                true
+            })
+        });
+
+        // Encrypting our Billing Address Details to be stored in Payment Intent
+        let key = key_store.key.get_inner().peek();
+        let billing_address_details = if details_present.is_some_and(|d| d.is_some_and(|d| d)) {
+            Some(raw_billing_address_details)
+                .as_ref()
+                .map(Encode::encode_to_value)
+                .transpose()
+                .change_context(errors::ApiErrorResponse::InvalidDataValue {
+                    field_name: "billing_address_details",
+                })
+                .attach_printable("Unable to convert billing address details to a value")?
+                .map(Secret::new)
+                .async_lift(|inner| encrypt_optional(inner, key))
+                .await
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Unable to encrypt guest customer details")?
+        } else {
+            None
+        };
```

Comment:
This will be handeled once this [PR](https://github.com/juspay/hyperswitch/pull/5090) is merged.